src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/DialogHelpers.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.IntegrationTest.Utilities;
+using Microsoft.VisualStudio.Threading;
+using UIAutomationClient;
+using AutomationElementIdentifiers = System.Windows.Automation.AutomationElementIdentifiers;
+using AutomationProperty = System.Windows.Automation.AutomationProperty;
+using ControlType = System.Windows.Automation.ControlType;
+
+namespace Roslyn.VisualStudio.IntegrationTests.InProcess
+{
+    public static class DialogHelpers
+    {
+        /// <summary>
+        /// Returns an <see cref="IUIAutomationElement"/> representing the open dialog with automation ID
+        /// <paramref name="dialogAutomationId"/>.
+        /// Throws an <see cref="InvalidOperationException"/> if an open dialog with that name cannot be
+        /// found.
+        /// </summary>
+        public static async Task<IUIAutomationElement> GetOpenDialogByIdAsync(JoinableTaskFactory joinableTaskFactory, IntPtr visualStudioHWnd, string dialogAutomationId, CancellationToken cancellationToken)
```

Comment:
📝 We should switch to WPF APIs for dialogs we own, and only use UI Automation for dialogs that might not be in WPF. That can be done later though.